### PR TITLE
Data from datasets in MemoryGraph

### LINKED ISSUE #289: View data fed from Arduino in Memorygraph
**Describe the issue**

We need to visualize the data stream coming from hardware. Can do that in MemoryGraph.

**Would you like to work on the issue?**

Issue for GSoC

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #289 

**Changes**: Graphing Data from datasets in MemoryGraph.

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [ ] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[app-debug.zip](https://github.com/fossasia/neurolab-android/files/3293224/app-debug.zip)

### CODE REVIEW

**Comment by jddeep** [comment]
@CloudyPadmal Please have a look. Can we have this have approach to get individual values from the dataset, which can then be plotted.

**Comment by CloudyPadmal** [comment]
Hi, rather than creating multiple usb instances, we'll create one instance and use it everywhere. Have a look at PSLab source code especially sciencelab and communication classes

**Comment by jddeep** [comment]
@CloudyPadmal Have a look at the current pattern. Can we follow this?

File: app/src/main/java/io/neurolab/program_modes/MemoryGraphActivity.java [hunk 1]
```diff
@@ -18,13 +28,71 @@
...
+            try {
+                data = new String(arg0, "UTF-8");
+                eegDataValues = parseDataForGraph(data);
+            } catch (UnsupportedEncodingException e) {
+                e.printStackTrace();
+            }
+        }
+    };
+    private final BroadcastReceiver broadcastReceiver = new BroadcastReceiver() { //Broadcast Receiver to automatically start and stop the Serial connection.
+        @Override
```
Comment (CloudyPadmal) [inline app/src/main/java/io/neurolab/program_modes/MemoryGraphActivity.java]: We need to move all connection related functions to entry point activity and let each UI related activities use them without reiterating same code

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/neurolab/program_modes/MemoryGraphActivity.java (hunk 1 above)
Comment (jddeep) [reply app/src/main/java/io/neurolab/program_modes/MemoryGraphActivity.java]: Yes @CloudyPadmal will direct to Launcher asap.

**Comment by jddeep** [comment]
@CloudyPadmal Please review this. Also as my Arduino isn't getting detected by my Android device (blocker), I couldn't test the device connection being detected in the phone and our app. Can you please test the APK and check whether the Arduino get's detected in out app (by that app bar icon change) and get me some feedback? Meanwhile I am trying to figure out why the board isn't getting detected.

**Comment by CloudyPadmal** [comment]
@jddeep sure..

**Comment by jddeep** [comment]
> @jddeep sure..

Thanks @CloudyPadmal Need it to be tested, to go forward in plotting the data.

**Review by CloudyPadmal — APPROVED** [review]
We'll proceed from here taking this as a base. Please resolve the conflicts

**Comment by jddeep** [comment]
@CloudyPadmal Conflicts resolved. Thanks

**Pull request merged**